### DISCUSSION TITLE
Custom mappings for keyboards without community layouts?

### DISCUSSION THREAD

**Discussion by TheRealMintd** [manna-harbour/miryoku#151] [Q&A] [answered]
I see the Kyria doesn't have any community layouts, and from what I gather from the documentation, I need to copy that in order to use the extra keys that I've got. What should I do for this case?

I've tried something like this, but it results in `macro "LAYOUT" requires 50 arguments, but only 48 given` :
```
#define MIRYOKU_MAPPING( \
      K00,  K01,  K02,  K03,  K04,         K05,  K06,  K07,  K08,  K09, \
      K10,  K11,  K12,  K13,  K14,         K15,  K16,  K17,  K18,  K19, \
      K20,  K21,  K22,  K23,  K24,         K25,  K26,  K27,  K28,  K29, \
      N30,  N31,  K32,  K33,  K34,         K35,  K36,  K37,  N38,  N39 \
) \
LAYOUT_split_3x6_5( \
XXX,  K00,  K01,  K02,  K03,  K04,                              K05,  K06,  K07,  K08,  K09,  XXX, \
XXX,  K10,  K11,  K12,  K13,  K14,                              K15,  K16,  K17,  K18,  K19,  XXX, \
XXX,  K20,  K21,  K22,  K23,  K24,  XXX,  XXX       XXX,  XXX,  K25,  K26,  K27,  K28,  K29,  XXX, \
                  XXX,  K32,  K33,  K34,  XXX       XXX,  K35,  K36,  K37,  XXX \
)
```

**Comment by manna-harbour** [comment, marked as answer]
The example in the docs happens to be for a community layout, but that's not required. For kyria, copy from https://github.com/manna-harbour/miryoku_qmk/blob/miryoku/keyboards/splitkb/kyria/keymaps/manna-harbour_miryoku/config.h.

**Reply by TheRealMintd** [reply]
Oh that worked, thank you!
